GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java

```diff
@@ -0,0 +1,236 @@
+        if (!granted) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION,
+                            Manifest.permission.ACCESS_COARSE_LOCATION},
+                    LOCATION_REQUEST_CODE);
+        }
+
+    }
+
+    private void requestReadWritePermissionsIfNotGranted() {
+        boolean granted = hasUserGrantedPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                && hasUserGrantedPermission(Manifest.permission.READ_EXTERNAL_STORAGE);
+        if (!granted) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                            Manifest.permission.READ_EXTERNAL_STORAGE},
+                    LOCATION_REQUEST_CODE);
+        }
+    }
+
+    /**
+     * Check if the user grant permissions required to run the app
+     *
+     * @param permissionName
+     * @return return true if permission granted and false if not
+     */
+    private boolean hasUserGrantedPermission(String permissionName) {
+        boolean granted = ActivityCompat.checkSelfPermission(this, permissionName)
+                == PackageManager.PERMISSION_GRANTED;
+        Log.d(TAG, "Permission " + permissionName + " : " + granted);
+        return granted;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions,
+            int[] grantResults) {
+        switch (requestCode) {
+            case LOCATION_REQUEST_CODE: {
+                if (grantResults.length <= 0
+                        || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                    Log.i(TAG, "Location Permissions are not granted.");
+                }
+            }
+            break;
+            case READ_WRITE_REQUEST_CODE: {
+                if (grantResults.length <= 0
+                        || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                    Log.i(TAG, "Read and Write Permission are not granted.");
+                }
+            }
+            default:
+                throw new IllegalStateException("Unexpected value: " + requestCode);
+        }
+    }
+
+    /**
+     * Set GPS if it's not enabled
+     */
+    private void setGpsEnabled() {
+        if (!isGpsEnabled) {
+            Intent settingsIntent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+            startActivity(settingsIntent);
+        }
+    }
+
+    /**
+     * Check GPS status
+     */
+    private void checkGpsStatus() {
+        isGpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+    }
+
+    /**
+     * On click of Start button, start capturing gps data
+     */
+    public void onStartButtonClick(View view) {
+        if (!isBound) {
+            Log.e(TAG, "GpsDataCaptureService is not bound, could not start capture.");
+            return;
+        }
```

Comment:
lets also add a check for for  GPS Status (checkGpsStatus)

Reply:
Done.

Reply:
Can you add check for  `Utils.isGpsEnabled(locationManager)` here  and show error on UI  if GPS is disabled?